astropy/io/fits/fitstime.py

```diff
@@ -175,22 +214,71 @@ def _verify_column_info(column_info, global_info):
             column_info['scale'] = global_info['scale']
             column_info['format'] = global_info.get('format', None)
 
-        # Non-linear coordinate types have "4-3" form and are not time coordinates
-        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+        else:
+            # Coordinate type is either an unrecognized local time scale
+            # or a linear coordinate type
             return False
 
-        # Coordinate type is either an unrecognized local time scale
-        # or a linear coordinate type
-        return False
+    elif (unit is not None and unit in FITS_TIME_UNIT) or location is not None:
+        column_info['scale'] = global_info['scale']
 
-    if global_info['scale'] is None:
+    else:
         return False
 
-    if (unit is not None and unit in FITS_TIME_UNIT) or location is not None:
-        column_info['scale'] = global_info['scale']
-        return True
+    if unit is not None:
+        column_info['format'] = UNIT_TO_FORMAT.get(unit, None)
```

Comment:
I think this will go wrong for a unit that is not in `UNIT_TO_FORMAT` such as 'h'. The right way would seem to just *use* the unit, i.e., convert the column values to a quantity with that unit. But as noted above, I actually don't see the need for this at all.